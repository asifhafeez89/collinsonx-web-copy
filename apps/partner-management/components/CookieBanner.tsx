import {
  Button,
  Flex,
  Text,
  Portal,
  Anchor,
} from '@collinsonx/design-system/core';
import styled from '@collinsonx/design-system/styled';
import { CONSENT } from 'config';
import Link from 'next/link';
import { useEffect, useState } from 'react';

const Container = styled.div`
  height: 92px;
  background-color: #f4f4f4;
  color: #000;
  box-shadow: 0px 2px 8px 0px rgba(0, 0, 0, 0.25);
  position: fixed;
  bottom: 0;
  width: 100%;

  /* freshchat widget zIndex = 2147483600 */
  z-index: 2147483601;
`;
const CookieBanner = () => {
  const [show, setShow] = useState(false);
  useEffect(() => {
    setShow(localStorage.getItem(CONSENT) !== 'accept');
  }, []);
  const handleClickAccept = () => {
    localStorage.setItem(CONSENT, 'accept');
    setShow(false);
  };
  return show ? (
    <Portal>
      <Container>
        <Flex h="100%" gap={48} align="center" justify="center">
          <Text size={18}>
            This website uses cookies to ensure you get the best experience.{' '}
            <Anchor
              component={Link}
              href="https://www.prioritypass.com/privacy-notice"
              sx={{
                fontWeight: 400,
                color: 'rgba(0, 0, 0, 1)',
              }}
            >
              Learn more
            </Anchor>
          </Text>
          <Button
            onClick={handleClickAccept}
            px={32}
            fw={700}
            sx={{
              background: 'rgba(109, 75, 246, 1)',
              fontSize: 18,
              height: 48,
            }}
          >
            Accept and close
          </Button>
        </Flex>
      </Container>
    </Portal>
  ) : null;
};

export default CookieBanner;
