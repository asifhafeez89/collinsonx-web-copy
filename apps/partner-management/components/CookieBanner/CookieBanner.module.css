.learnMoreLink {
  font-weight: 400;
  color: #000;
}
.acceptBtn {
  background: #6d4bf6;
  font-size: rem(18px);
  height: rem(48px);
}
