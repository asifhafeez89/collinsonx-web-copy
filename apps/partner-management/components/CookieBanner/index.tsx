import {
  Button,
  Flex,
  Text,
  Portal,
  Anchor,
} from '@collinsonx/design-system/core';
import styled from '@collinsonx/design-system/styled';
import { CONSENT } from 'config';
import Link from 'next/link';
import { useEffect, useState } from 'react';
import classes from './CookieBanner.module.css';

const Container = styled.div`
  height: 92px;
  background-color: #f4f4f4;
  color: #000;
  box-shadow: 0px 2px 8px 0px rgba(0, 0, 0, 0.25);
  position: fixed;
  bottom: 0;
  width: 100%;

  /* freshchat widget zIndex = 2147483600 */
  z-index: 2147483601;
`;
const CookieBanner = () => {
  const [show, setShow] = useState(false);
  useEffect(() => {
    setShow(localStorage.getItem(CONSENT) !== 'accept');
  }, []);
  const handleClickAccept = () => {
    localStorage.setItem(CONSENT, 'accept');
    setShow(false);
  };
  return show ? (
    <Portal>
      <Container>
        <Flex h="100%" gap={48} align="center" justify="center">
          <Text style={{ fontSize: 18 }}>
            This website uses cookies to ensure you get the best experience.{' '}
            <Anchor
              className={classes.learnMoreLink}
              component={Link}
              href="https://www.prioritypass.com/privacy-notice"
            >
              Learn more
            </Anchor>
          </Text>
          <Button
            className={classes.acceptBtn}
            onClick={handleClickAccept}
            px={32}
            fw={700}
          >
            Accept and close
          </Button>
        </Flex>
      </Container>
    </Portal>
  ) : null;
};

export default CookieBanner;
