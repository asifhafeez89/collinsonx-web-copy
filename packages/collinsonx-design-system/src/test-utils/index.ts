import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
