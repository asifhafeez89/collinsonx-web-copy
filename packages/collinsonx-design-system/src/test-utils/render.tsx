import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
