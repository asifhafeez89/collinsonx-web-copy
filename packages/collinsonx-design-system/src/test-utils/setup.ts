import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
