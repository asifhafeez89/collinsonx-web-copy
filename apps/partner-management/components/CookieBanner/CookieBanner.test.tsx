import CookieBanner from './';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render } from '@collinsonx/design-system/test-utils';

describe('<CookieBanner />', () => {
  it('should render', () => {
    const component = render(<CookieBanner />);

    expect(component).toMatchSnapshot();
  });
  it('button click', () => {
    const component = render(<CookieBanner />);
    const btn = component.getByRole('button');
    fireEvent.click(btn);
  });
});
