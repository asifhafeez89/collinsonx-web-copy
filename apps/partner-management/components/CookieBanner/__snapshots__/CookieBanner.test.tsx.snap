// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CookieBanner /> should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  height: 92px;
  background-color: #f4f4f4;
  color: #000;
  box-shadow: 0px 2px 8px 0px rgba(0, 0, 0, 0.25);
  position: fixed;
  bottom: 0;
  width: 100%;
  z-index: 2147483601;
}

<body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
    </div>
    <div
      data-portal="true"
    >
      <div
        class="emotion-0"
      >
        <div
          class="m-8bffd616 mantine-Flex-root __m__-r0"
          style="gap: calc(3rem * var(--mantine-scale)); align-items: center; justify-content: center; height: 100%;"
        >
          <p
            class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
            style="font-size: 18px;"
          >
            This website uses cookies to ensure you get the best experience.
             
            <a
              class="mantine-focus-auto m-849cf0da learnMoreLink m-b6d8b162 mantine-Text-root mantine-Anchor-root"
              data-underline="hover"
              href="https://www.prioritypass.com/privacy-notice"
            >
              Learn more
            </a>
          </p>
          <button
            class="mantine-focus-auto mantine-active acceptBtn m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
            style="font-weight: 700;"
            type="button"
          >
            <span
              aria-hidden="true"
              class="m-a25b86ee mantine-Button-loader"
            >
              <span
                class="m-b34414df m-5ae2e3c mantine-Loader-root"
                style="--loader-size: calc(var(--button-height) / 1.8); --loader-color: var(--button-color);"
              />
            </span>
            <span
              class="m-80f1301b mantine-Button-inner"
            >
              <span
                class="m-811560b9 mantine-Button-label"
              >
                Accept and close
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
